### ISSUE TITLE
Should we remove mypy type checking or validation

### ISSUE THREAD

**Issue by amimas** [gitlabform/gitlabform#586]
Frankly I am fed up with constant mypy issues. I don’t recall any case where it helped while I remember many ones where it was just an nuisance. I propose to get rid of this check completely. Any objections, @amimas ?

_Originally posted by @gdubicki in https://github.com/gitlabform/gitlabform/pull/585#discussion_r1284414736_

**Comment by amimas** [comment]
I like the concept or idea of type checking and validation. But, I'm really new to Python and don't have enough experience to suggest whether to keep using `mypy` for this project or not. I'd go with your expertise @gdubicki . Hopefully we hear from other experienced contributors as well.

**Comment by adam-moss** [comment]
I use type checking, and `mypy`, generally without issue, with the type annotation settings turned on in vs-code and a bit of policing from `ruff`.

I'd be more inclined to use `pydantic` on anything new though.

**Comment by gdubicki** [comment]
Isn’t pydantic use-case a bit different, @adam-moss? I haven’t used it but based on what I read is more of a data validation solution than a type-checker.

**Comment by jimisola** [comment]
I don't think it should be removed. Use it extensively at work. It has several pros.

There was a survey made about typing in python (Pylance team at Microsoft and PyCharm at JetBrains):

Discussion: https://discuss.python.org/t/2024-python-typing-survey-analysis/61456
Results (check out the slides): https://discuss.python.org/t/python-typing-survey-results/71821
